kiali.go

```diff
@@ -115,6 +122,36 @@ func main() {
 	server.Stop()
 }
 
+func waitForSecret() {
+	foundSecretChan := make(chan security.Credentials)
+	go func() {
+		errs := 0
+		for true {
+			username, uErr := ioutil.ReadFile(config.LoginSecretUsername)
+			passphrase, pErr := ioutil.ReadFile(config.LoginSecretPassphrase)
+			if uErr == nil && pErr == nil {
+				if string(username) != "" && string(passphrase) != "" {
+					log.Info("Secret is now available.")
+					foundSecretChan <- security.Credentials{
+						Username:   string(username),
+						Passphrase: string(passphrase),
+					}
+					break
+				}
+			}
+			errs++
+			if (errs % 5) == 0 {
+				log.Warning("Kiali is missing a secret that contains both 'username' and 'passphrase'")
+			}
+			time.Sleep(2 * time.Second)
+		}
+	}()
+	secret := <-foundSecretChan
+	config.Get().Server.Credentials.Username = secret.Username
+	config.Get().Server.Credentials.Passphrase = secret.Passphrase
+	config.Get().Server.Credentials.AllowAnonymous = false
+}
+
```

Comment:
Won't this block pod termination if scaled down or deleted before a secret is in place?

Reply:
good question. I will test that.

Reply:
This will not block pod termination. I just tested it. If Kiali is sitting waiting for a secret, and you delete the pod, the pod will terminate and a new pod will take its place normally. I also went to the kiali deployment and scaled down to 0 while kiali was waiting for a secret and it successfully scaled down. I then scaled back up to 1 and it came back up (but of course it again was waiting for a secret). 

I saw no issues either way.